### ISSUE TITLE
Change of proximity pattern

### ISSUE THREAD

**Issue by Meisterschueler** [svoop/ogn_client-ruby#3]
https://github.com/svoop/ogn_client-ruby/blob/e0f28952a42deccb764a81da009262c0c4af5b49/lib/ogn_client/messages/sender_beacon.rb#L18

Since we can receive multiple "hear..." messages ("hear1234 hearAFFE, hear1A2B") I would change the pattern from 
`(?:hear(?P<proximity>.+))?`
to
`(?:(?P<proximity>(hear[\dA-F]{4}\s?)+))?`

**Comment by svoop** [comment]
Oops, you're right, I'll fix that. Thanks for the hint!

**Comment by svoop** [comment]
@Meisterschueler 

Off topic, but what are you making of these messages containing `OGNTRK` instead of `APRS`which appear every now and then?

    OGN395F39>OGNTRK,qAS,Barton:/230738h5145.94N/00111.50W'000/000/A=000279 !W58! id07395F39 +000fpm +0.0rot FL002.03 46.0dB 0e +3.6kHz gps3x5

I tend to change the pattern to accept anything in the receiver part (after the >), but it would be nice to talk to the sender of these beacons (Barton) beforehand. Unfortunately, I don't find any details about who operates it.

**Comment by svoop** [comment]
@Meisterschueler 

You're right about the pattern, however, further down in the parser, I do handle multiple `hear`s by splitting the matched string, e.g:

Given `hearAAAA hearBBBB`, the current pattern `(?:hear(?<proximity>.+))?` is matching  `AAAA hearBBBB`. In the setter method, this is split with `split(/\shear/)` which results in the array `["AAAA", "BBBB"]`.

Now, this works, but it's not really safe should the protocol amend further pieces of information after the `hear`s. Do you know whether this is a likely scenario at all?

**Comment by Meisterschueler** [comment]
For the OGNTRK I found this:
https://github.com/gewesp/cpp-lib/commit/d22422cff6e7ed69d973158fcaea2c2d5fbe4948#diff-2d1ffa5e60521954c8b1cbe7c803babe

**Comment by svoop** [comment]
@Meisterschueler Ah, interesting bit, apparently, there are new data sources with potentially diverging message formats coming up. I've [asked about this development here](https://github.com/gewesp/cpp-lib/commit/d22422cff6e7ed69d973158fcaea2c2d5fbe4948#diff-699e0f6e59fd8be4382481122cdd0a7a). The thread is worth a look!
